From 3a52125f81d28fcad5d3c841b94a3e4bad295d9e Mon Sep 17 00:00:00 2001
From: Tim Kosse <tim.kosse@filezilla-project.org>
Date: Sun, 20 Mar 2016 10:22:13 +0100
Subject: [PATCH] Fix wxTopLevelWindowGTK::DoSetSizeHints if a window scale
 factor larger than 1  is set.

In this case, passing INT_MAX as maximum width/height hint
to gtk_window_set_geometry_hints leads to integer overlows.

Instead, use INT_MAX / 16 which should work with all
resonable scale factors.
---
 src/gtk/toplevel.cpp | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/src/gtk/toplevel.cpp b/src/gtk/toplevel.cpp
index 13dc40a..1f054d7 100644
--- a/src/gtk/toplevel.cpp
+++ b/src/gtk/toplevel.cpp
@@ -1229,8 +1229,12 @@ void wxTopLevelWindowGTK::DoSetSizeHints( int minW, int minH,
     int hints_mask = GDK_HINT_MIN_SIZE | GDK_HINT_MAX_SIZE;
     hints.min_width = 1;
     hints.min_height = 1;
-    hints.max_width = INT_MAX;
-    hints.max_height = INT_MAX;
+    // Due to HiDPI UI scale, using INT_MAX leads to integer
+    // overflows when calculating the actual window size.
+    // Divide by 16 to to accomodate crazy high scale factors
+    // without overflowing.
+    hints.max_width = INT_MAX / 16;
+    hints.max_height = INT_MAX / 16;
     int decorSize_x;
     int decorSize_y;
 #ifdef HAS_CLIENT_DECOR
-- 
2.5.0

